Enhance time formatting with new hide options

gerbermichi opened the pull request:
Added options `hideYears`, `hideYearsAndDays`, and `hideSeconds` to allow for more customizable time formatting.

Fixes #70

gerbermichi committed:
Enhance time parsing with new hide options

Added options `hideYears`, `hideYearsAndDays`, and `hideSeconds` to allow for more customizable time formatting.

sindresorhus commented on index.d.ts:
```diff
@@ -75,6 +75,27 @@ export type Options = {
+
+	 @default false
+	 */
+	readonly hideYears?: boolean;
```
Should be singular, as it should be read as "hide year component".

sindresorhus commented on index.d.ts:
```diff
@@ -75,6 +75,27 @@ export type Options = {
+	 Hides the years and shows the hidden years additionally as days (365 per year).: `1y 3d 5h 1m 45s` → `368d 5h 1m 45s`.
+
+	 @default false
+	 */
```
Incorrect indentation.

sindresorhus commented on test.js:
```diff
@@ -379,6 +379,46 @@ runTests({
+		[(1000 * 60 * 67 * 24 * 465) + (1000 * 60 * 60 * 15) + (1000 * 60 * 59) + (1000 * 59) + 543, {hideSeconds: true, hideYearsAndDays: true, colonNotation: true}, '12477:59'],
+	],
+});
+
```
I would like to see some more tests on how to works together with the other options.

gerbermichi replied:
Hi, I added numerous test cases. Do you have any specific in mind?

sindresorhus commented:
Thanks for contributing.

sindresorhus commented:
The readme needs to be updated too.

gerbermichi committed:
Rename hideYears and hideYearsAndDays options

Updated options "hideYears" to "hideYear" and "hideYearsAndDays" to "hideYearAndDays" for consistency. Also updated relevant code and tests to use the new option names.

gerbermichi committed:
Add new time formatting options to the readme.md

gerbermichi committed:
Refactor to use BigInt for hours calculation

Changed the hours calculation to use BigInt to handle large integers accurately. This ensures that the computation remains precise even when dealing with large numbers, enhancing the robustness of the code.

gerbermichi committed:
Add new test cases for enhanced time formatting options

Included various configurations to test detailed time formatting including decimal seconds, sub-millisecond precision, and verbose modes. This ensures comprehensive coverage across different time display preferences.

gerbermichi committed:
Fix formatting in TypeScript definition comments

Removed extra spaces in the comment blocks and aligned the `@default` lines correctly. This change improves code readability and consistency in the index.d.ts file.

gerbermichi committed:
Reformat test cases in test.js

sindresorhus committed:
Update index.d.ts

sindresorhus committed:
Update index.js

sindresorhus committed:
Update readme.md
